sql/parser/eval.go

```diff
@@ -667,23 +686,41 @@ func evalFuncExpr(expr *FuncExpr) (Datum, error) {
 		types = append(types, reflect.TypeOf(arg))
 	}
 
-	var impl func(DTuple) (Datum, error)
-	for _, candidate := range candidates {
-		if candidate.match(types) {
-			impl = candidate.fn
-			break
+	needMatch := true
```

Comment:
Yeah, it was just a small optimization to avoid performing the match a second time when not necessary. Removed for now.